interfaces/builtin/registry.go

```diff
@@ -0,0 +1,107 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/registry"
+	"github.com/snapcore/snapd/snap"
+)
+
+// The registry interface can be plugged by snaps that want to access specific
+// registry views. Plugs are auto-connected if the registry's account is the
+// same as the snap's publisher.
+const registrySummary = `allows accessing a registry through a view`
+
+const registryBaseDeclarationSlots = `
+  registry:
+    allow-installation:
+      slot-snap-type:
+        - core
+`
+const registryBaseDeclarationPlugs = `
+  registry:
+    allow-auto-connection:
+      plug-attributes:
+        account: $PLUG_PUBLISHER_ID
+`
+
+// registryInterface allows accessing a registry through a view
+type registryInterface struct {
+	commonInterface
+}
+
+func (iface *registryInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	account, ok := plug.Attrs["account"].(string)
+	if !ok || account == "" {
+		return fmt.Errorf(`registry plug must have an "account" attribute`)
+	}
+
+	view, ok := plug.Attrs["view"].(string)
+	if !ok || view == "" {
+		return fmt.Errorf(`registry plug must have a "view" attribute`)
+	}
+
+	if err := validateView(view); err != nil {
+		return err
+	}
+
+	role, ok := plug.Attrs["role"].(string)
+	if !ok {
+		// defaults to manager
+		role = "manager"
+		plug.Attrs["role"] = role
```

Comment:
we probably want to double check this

Reply:
so result of the discussion is that there should not be a default for this

Reply:
I've changed this now